src/tqec/circuit/__init__.py

```diff
@@ -1,5 +1,4 @@
-"""Defines core classes and functions to represent and manipulate quantum
-circuits.
+"""Defines core classes and functions to represent and manipulate quantum circuits.
```

Comment:
There is a significant overlap with imperative mood in one of the other open PRs. I think it would be better if we merge that one first and then fix the errors in this PR.

Reply:
The PRs are chained: `main` -> `ci/fix_d205_errors` (#638) -> `ci/pyupgrade_and_rust` (#640) -> `ci/ruff_specific_rules` (#641) -> `ci/add_pylint_through_ruff` (#642) -> `ci/undocumented_params` (#643).

I plan to merge them one by one, starting by the closest one to `main` (#638). That will incur a few merges because we stash commits, but I am willing to do the merges to keep the history clean.